enhancements/two-nodes-openshift/2no.md

```diff
@@ -113,32 +119,57 @@ In both cases, the control-plane's dependence on etcd will cause it to respond w
-1. Should fencing be made active during the installation or should pacemaker start with it disabled and only enable it after being signaled by the in-cluster entity when the cluster installation is detected as successful?
-2. What mechanism will pacemaker use to get access to the secret linked from the BareMetalHost CRD?
+###### Configuring Fencing Via 2NO Resource Agent Remediation Operator
+Fencing setup is the last important aspect of the cluster installation. For the cluster installation to be successful, fencing should be configured and active before we declare the installation successful. In order to do this, baseboard management console (BMC) credentials need to be made available to the control-plane nodes as part of pacemaker initialization.
+To ensure rapid fencing using pacemaker, we will collect RedFish details (address, username, and **password**) for each node via the install-config.
+This will take a format similar to that of the [Baremetal Operator](https://docs.openshift.com/container-platform/4.17/installing/installing_bare_metal_ipi/ipi-install-installation-workflow.html#bmc-addressing_ipi-install-installation-workflow).
+We will create a new CRD that stores the BMC address as well as a credentialsName to be managed by the new operator. This will resemble the BMC specification used by the [BareMetalHost](https://docs.openshift.com/container-platform/4.17/rest_api/provisioning_apis/baremetalhost-metal3-io-v1alpha1.html#spec-bmc) CRD.
 
-If available via the SaaS offering (not confirmed), ZTP may be evaluated as a future offering. This will need further evaluation to ensure passwords are appropriately handled.
+BMC information can be used to change the power state of a baremetal machine, so it's critically important that we ensure that pacemaker is the **only entity** responsible for these operations to prevent conflicting requests to change the machine state.
+This means that we need to ensure that there are protections in the Baremetal Operator (BMO) to prevent control plane nodes from having power management enabled in a two-node topology. Additionally, optional operators like Node Health Check, Self Node Remediation, and Fence Agents Remediation must have the same considerations but these should not be present during installation.
 
-Everything else about cluster creation will be an opaque implementation detail not exposed to the user.
+For a two-node cluster to be successful, we need to ensure the following:
+1. The new operator must be able to watch for changes to the new CRD and synchronize the BMC secrets for RHEL-HA once during initialization and again whenever the CRD is updated.
```

Comment:
I might be wrong, but IIRC this part will is already covered by pacemaker (i.e making sure BMC credentials are synced) 
@fabbione please keep me honest here :sweat_smile:

Reply:
@mshitrit it depends how fencing is configured in pacemaker itself. Usually there 3 ways to manage fencing credentials in pacemaker for fencing.

1. username/password are given to pacemaker via pcs configuration. In this case if any of those are changed, a pcs config update is required. pacemaker will take care to synchronize the configuration change across all nodes of the cluster.
2. password is stored in a file on disk. In this case the operator will update the file and pacemaker/fence agent will pick it up automatically on the next run.
3. password-script, a shell/python/$interpreter script that will get the credentials somehow/somewhere. Same as point 2, it will be executed each time a fencing operation is required.

For now, we should stick to either 1 or 2. Grabbing credentials in 3 can be challenging depending on failure scenarios.

Reply:
I decided to go into less detail than I originally intended. Just noted that the secrets are on disk via MCO to be consumed by pacemaker however we decide to do that.